packages/parity-electron/src/isParityRunning.ts

```diff
@@ -6,25 +6,26 @@
 import axios from 'axios';
 import * as retry from 'async-retry';
 
-import { cli } from './utils/cli';
 import logger from './utils/logger';
 
-/**
- * Try to ping these hosts to test if Parity is running.
- */
-const hostsToPing = ['http://127.0.0.1:8545', 'http://127.0.0.1:8546'];
-if (cli.wsInterface || cli.wsPort) {
-  // Also try custom host/port if a --ws-interface or --ws-port flag is passed
-  hostsToPing.push(
-    `http://${cli.wsInterface || '127.0.0.1'}:${cli.wsPort || '8546'}`
-  );
-}
-
 /**
  * Detect if another instance of parity is already running or not. To achieve
  * that, we just ping on the common hosts, see {@link hostsToPing} array.
  */
-export const isParityRunning = async () => {
+export const isParityRunning = async ({
+  wsInterface = '127.0.0.1',
+  wsPort = '8546'
+}: {
+    wsInterface: string;
+    wsPort: string;
+  }) => {
+  /**
+   * Try to ping these hosts to test if Parity is running.
+   */
+  const hostsToPing = [
+    `http://${wsInterface}:${wsPort}`
```

Comment:
What if the user is running Parity with ws server disabled? E.g. people who create ssh tunnels only on 8545.

I'd suggest always pinging `http://127.0.0.1:8545` and `http://127.0.0.1:8546` first. Debatable though.

Reply:
Does Fether still work with Parity if it has ws server disabled? does Fether then fallback to http jsonrpc?

Reply:
Fether will show a loading screen "Connecting to Parity...".

However it's still better than running an undesirable instance of Parity, or showing an error message.